### PR TITLE
[test] [sagemaker] Fix sagemaker remote test reports

### PR DESCRIPTION
**Pull request by saimidu**
*Issue #, if available:*

## Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [x] (If applicable) I've documented below the tests I've run on the DLC image
- [x] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [x] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
SageMaker Remote Tests would sometimes fail on CI jobs, but the CI check would still report a Success.
The cause was identified as the following two issues:
1. The lack of an Exception or Error raised by the pytest command failure, because the fabric `run` command is run with `warn=True` to report the results in metrics.
This was fixed by adding an assert at the end of the test to check that the return code is 0, and raise an error otherwise.
2. The failed test has its report overwritten by a non-failing test report being written into the same filename. This happened because the training and inference tests were both executed simultaneously, and both wrote their results into the same file name, for a single image type.
This was fixed by adding job-type to the pytest report file name.

*Tests run:*
1. The first set of CI tests on this PR was run with the pytest command set to deliberately fail, with `pytest xyz`. This resulted in a test failure, but a successful CI check on the PR.
2. The second set of CI tests on this PR was run with an assert after the deliberate failure, where the return code was checked, and an AssertionError was thrown. This caused the CI check on the PR to report a Failure.
3. The final state of the PR has the wrong pytest command removed, and the AssertionError triggered on actually failing tests.

*DLC image/dockerfile:*
N/A

*Additional context:*
N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by arjkesh** [comment]
Can you please update the description and provide an example failure?

File: test/test_utils/sagemaker.py [hunk 1]
```diff
@@ -267,6 +267,7 @@ def execute_sagemaker_remote_tests(image):
             context.run("pip install -r requirements.txt", warn=True)
             res = context.run(pytest_command, warn=True)
             metrics_utils.send_test_result_metrics(res.return_code)
+            assert res.ok, f"{pytest_command} failed."
```
[the remark this answers is not in the document]
Comment (saimidu) [reply test/test_utils/sagemaker.py:270]: Propagate as in print out the exact error message? No, you would just see `"<pytest command> failed"`. But the way we execute SageMaker remote tests prints all the errors/logs out to stdout/stderr anyway.

**Review by SergTogul — DISMISSED** [review]

**Comment by tejaschumbalkar** [comment]
LGTM. Will approve once the build config changes are reverted.

File: test/test_utils/sagemaker.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (arjkesh) [reply test/test_utils/sagemaker.py:270]: The error that ultimately gets propagated in this case is an assertion error that provides the pytest command but no other context.

Let's raise a) a more appropriate error, can be a custom error like DLCSageMakerTestFailure. We can include pytest command, which is important, as well as other relevant info we have access to, like return code, and a comment to check reports or logs for more details.

[the remark this answers is not in the document]
Comment (saimidu) [reply test/test_utils/sagemaker.py]: Thanks. Improved the exception raised, and added more details to the error message.

**Review by tejaschumbalkar — APPROVED** [review]

File: test/test_utils/sagemaker.py [hunk 2]
```diff
@@ -267,6 +271,11 @@ def execute_sagemaker_remote_tests(image):
             context.run("pip install -r requirements.txt", warn=True)
             res = context.run(pytest_command, warn=True)
             metrics_utils.send_test_result_metrics(res.return_code)
+            if res.failed:
+                raise DLCSageMakerRemoteTestFailure(
+                    f"{pytest_command} failed with error code: {res.return_code}\n"
+                    f"Traceback:\n{res.stdout}"
```
Comment (arjkesh) [inline test/test_utils/sagemaker.py:277]: the "Traceback" part here doesn't appear to print out anything. Perhaps this needs to print out stderr? It looks like relevant debug information is not suppressed, so I am ok with removing this line

Comment (saimidu) [reply test/test_utils/sagemaker.py:277]: Shall I fast follow with another PR?

**Pull request merged**